governance/api.go

```diff
@@ -165,6 +165,18 @@ func (api *PublicGovernanceAPI) ItemsAt(num *rpc.BlockNumber) (map[string]interf
 	}
 }
 
+func (api *PublicGovernanceAPI) GetStakingInfo(num *rpc.BlockNumber) (*reward.StakingInfo, error) {
+	blockNumber := uint64(0)
+	if num == nil || *num == rpc.LatestBlockNumber {
+		blockNumber = api.governance.blockChain.CurrentHeader().Number.Uint64()
+	} else if *num == rpc.PendingBlockNumber {
+		return nil, kerrors.ErrPendingBlockNotSupported
+	} else {
+		blockNumber = uint64(num.Int64())
+	}
+	return reward.GetStakingInfo(blockNumber), nil
```

Comment:
There are many repeated code. I think it can be defined as a function.

Reply:
Right. This is not only repeated one. Can we update this later with other repeated code fragments?

Reply:
OK.